apps/editing-toolkit/editing-toolkit-plugin/wpcom-block-editor-nux/src/welcome-tour/utils.ts

```diff
@@ -0,0 +1,20 @@
+/**
+ * External Dependencies
+ */
+import { useMobileBreakpoint } from '@automattic/viewport-react';
+import type { Step, Steps } from '@automattic/tour-kit';
+
+export function usePrefetchTourAssets( steps: Step | Steps ): void {
```

Comment:
Why not keep the signature `( steps: Steps ) => void` and always pass in an array? I don't think we need the extra complexity with dealing with both singular and multiple steps, and we can also keep the semantics sane ("steps" being plural).

Reply:
Thanks for addressing this. 

In `apps/editing-toolkit/editing-toolkit-plugin/wpcom-block-editor-nux/src/welcome-tour/tour-launch.js:30`, we pass a singular argument to the hook, hence we have to add additional logic to diverge between singular and multiple arguments. 

```js
// Preload first card image (others preloaded after open state confirmed)
new window.Image().src = usePrefetchTourAssets( getTourSteps( localeSlug )[ 0 ] );
```

It feels wrong to wrap a singular argument in an array just to satisfy a hook's signature but doing it makes the hook:
1. More comprehensible.
2. More performant.
3. Easier to reuse.

Therefore I refactored the code in fba3324.